src/index.js

```diff
@@ -30,4 +29,82 @@ const csvDownload = (data, name, delimiter) => {
   document.body.removeChild(link)
 }
 
-export default csvDownload
+/**
+ * Provides support for digesting and exporting JSON to CSV
+ * 
+ * @param {JSON} __data 
+ * @param {String} __name 
+ * @param {Char} __delimiter 
+ * @returns Object
+ */
+export const jsonToCsv = (__data, __name, __delimiter) => {
+  const data = __data
+  const filename = __name || `export.csv`
+  const delimiter = __delimiter || `,`
+
+  const header = Array.from(
+    new Set(data.reduce((r, e) => [...r, ...Object.keys(e)], []))
+  )
+
+  let csv = data.map(row => header.map(fieldName => JSON.stringify(row[fieldName] || '')).join(delimiter))
+
+  csv.unshift(header.join(delimiter))
+  csv = csv.join('\r\n')
+
+  return {
+    filename: filename,
+    object: csv
+  }
+}
+
+/**
+ * This provides support for digesting and exporting CSV data to JSON.
+ *  
+ * @param {String} __data 
+ * @param {String} __name 
+ * @param {Char} __delimiter 
+ */
+export const csvToJson = (__data, __name, __delimiter) => {
+  let returnData = {
+    filename: null,
+    object: null,
+    error: null
+  }
```

Comment:
minor: could you rewrite these new functions to be without reassignments? I prefer the code style here to be functional.